server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -837,6 +861,45 @@ private IndexMetadata updateIndexSettings(
                         return builder.settings(settingsBuilder).build();
                     }
 
+                    private void validateSearchableSnapshotRestorable(long totalRestorableRemoteIndexesSize) {
+                        ClusterInfo clusterInfo = clusterInfoSupplier.get();
+                        double remoteDataToFileCacheRatio = DATA_TO_FILE_CACHE_SIZE_RATIO_SETTING.get(clusterService.getSettings());
+                        Map<String, FileCacheStats> nodeFileCacheStats = clusterInfo.getNodeFileCacheStats();
+                        if (nodeFileCacheStats.isEmpty() || remoteDataToFileCacheRatio <= 0.0f) {
```

Comment:
Floating point math can be weird. Is there any risk of a setting value of `0` resulting in a double with a value very very close, but slightly larger than, zero? We could guard against that by making the comparison `<= 0.01` or something like that.

Reply:
Added a slight buffer in.